pk3DS/Main.cs

```diff
@@ -1,1369 +1,1387 @@
+        }
+
+        private void L_Game_Click(object sender, EventArgs e) => new EnhancedRestore(Config).ShowDialog();
+
+        private void B_Open_Click(object sender, EventArgs e)
+        {
+            FolderBrowserDialog fbd = new FolderBrowserDialog();
+            if (fbd.ShowDialog() == DialogResult.OK)
+                openQuick(fbd.SelectedPath);
+        }
+
+        private void ChangeLanguage(object sender, EventArgs e)
+        {
+            if (InvokeRequired)
+                Invoke((MethodInvoker)delegate { Language = CB_Lang.SelectedIndex; });
+            else Language = CB_Lang.SelectedIndex;
+            if (Config != null)
+                Config.Language = Language;
+            Menu_Options.DropDown.Close();
+            if (!Tab_RomFS.Enabled || Config == null)
+                return;
+
+            if ((Config.XY || Config.ORAS) && Language > 7)
+            {
+                WinFormsUtil.Alert("Language not available for games. Defaulting to English.");
+                if (InvokeRequired)
+                    Invoke((MethodInvoker)delegate { CB_Lang.SelectedIndex = 2; });
+                else CB_Lang.SelectedIndex = 2;
+                return; // set event re-triggers this method
+            }
+
+            updateGameInfo();
+            Config.InitializeGameText();
+            Properties.Settings.Default.Language = Language;
+            Properties.Settings.Default.Save();
+        }
+
+        private void Menu_Exit_Click(object sender, EventArgs e)
+        {
+            Close();
+        }
+
+        private void CloseForm(object sender, FormClosingEventArgs e)
+        {
+            if (Config == null)
+                return;
+            var g = Config.GARCGameText;
+            string[][] files = Config.GameTextStrings;
+            g.Files = files.Select(x => TextFile.getBytes(Config, x)).ToArray();
+            g.Save();
+
+            try
+            {
+                File.WriteAllLines(RandSettings.FileName, RandSettings.Save(), Encoding.Unicode);
+            }
+            catch { }
+        }
+
+        private void openQuick(string path)
+        {
+            if (threadActive())
+                return;
+
+            if (!Directory.Exists(path)) // File
+            {
+                if (!File.Exists(path))
+                    return;
+
+                FileInfo fi = new FileInfo(path);
+                if (fi.Name.Contains("code.bin")) // Compress/Decompress .code.bin
+                {
+                    if (fi.Length % 0x200 == 0 && WinFormsUtil.Prompt(MessageBoxButtons.YesNo, "Detected Decompressed code.bin.", "Compress? File will be replaced.") == DialogResult.Yes)
+                        new Thread(() => { threads++; new BLZCoder(new[] { "-en", path }, pBar1); threads--; WinFormsUtil.Alert("Compressed!"); }).Start();
+                    else if (WinFormsUtil.Prompt(MessageBoxButtons.YesNo, "Detected Compressed code.bin.", "Decompress? File 
```

Comment:
Is this event handler being called asynchronously? I don't think this Invoke is necessary as long as the caller is on the UI thread.